source/develop/release-management/features/network/manageiq_ovn.html.md

```diff
@@ -0,0 +1,140 @@
+  * In ManageIQ the 'services' URLs are not supplied by the user but determined from the Keystone. When Keystone is asked for the `token` it replies with the `token` and a `service_catalog` with all the services information (URLs, credentials, etc).
+* Tenants
+  * Neutron supports tenants.
+  * oVirt's OVN provider doesn't.
+  * ManageIQ expects the Neutron network provider to have tenants.
+* oVirt's OVN provider has 'v2' Keystone API. However, 'v2' API is already considered as legacy.
+
+
+### Solving the gaps
+* Communication and authentication
+ * Creating network manager class that doesn't inherit from the OpenStack one.
+ * oVirt's network manager will take responsibility for its communication and authentication reusing the communication and authentication code of the OpenStack infra manager.
+ * The collector, parser, refresher and entities classes (cloud_network, cloud_subnet, network_port etc) inherit from the OpenStack ones.
+ * Upon each update of the oVirt's infra manager
+   * all the network providers will be retrieved from oVirt.
+   * only the first network provider will be processed.
+   * if a network manager doesn't exist it will be created.
+   * the network's provider `authentication_url` will be parsed and the network manager will be updated accordingly. The fields that will be updated are: `hostname`, `port`, `api_version` and `security_protocol` ('ssl' for 'https' and 'non-ssl' for 'http').
+* Service Catalog
+  * oVirt's OVN provider was updated to send also `service_catalog` when it is asked for a `token`. Only the relevant services will be in the catalog - 'identity' (keystone), 'network' and 'compute' (see open questions).
+* Tenants (there are two possible solutions)
+  1. oVirt OVN provider keystone will send empty tenants list when it is asked for tenants.
+    * ManageIQ code that assumes there are tenants will be overridden for the oVirt's network provider.
+    * Currently the only code I found that need to be overridden is the ui code for adding/updating the entities (networks, subnets, etc), which has the tenants as a mandatory field.
+  2. oVirt provider OVN will report a dummy tenant that will be stored in the 'cloud_tenant' table of ManageIQ.
+    * When creating the network manager entities, the user will have to choose this tenant (which has no sense).
+      * Maybe we can automatically choose the dummy tenant and hide the field from the user.
+
+## Open Questions
+* `ManageIQ::Providers::Openstack::legacy:OpenstackHandle` is responsible to communicate with the openstack-fog. It is placed in a 'legacy' directory. Why? Is it going to be replaced soon?
+* Why does the `openstack_handle/handle.accessible_tenants` method first tries to connect to the 'compute' service?
+  * Current workaround - the oVirt OVN provider reports the 'compute' service in the service catalog with the same URL and credentials as the network service.
```

Comment:
not sure where, but please add:
when an operator adds an OVN network via ManageIQ, MIQ would issue an auto-import of the new network into oVirt.

Reply:
added to follow-up features.